### ISSUE TITLE
Rule in exec-once doesn't apply

### ISSUE THREAD

**Issue by diniamo** [hyprwm/Hyprland#6569]
### Bug or Regression?

Bug

### System Info and Version

<details> <summary>System/Version info</summary>

```sh
Hyprland, built from branch  at commit 1360677478e867034713e5d43d7a8a8f6bf1343d  ().
Date: 2024-06-17
Tag: , commits: 1360677478e867034713e5d43d7a8a8f6bf1343d

flags: (if any)
```

</details>

### Description

Extra rules (or at least this one) don't apply in `exec-once`.

Note that the same rhs works with `hyprctl dispatch exec`.

### How to reproduce

1. Put `exec-once=[workspace special:terminal silent] <terminal>`
2. Restart Hyprland
3. Terminal spawns in normal workspace

**Comment by diniamo** [comment]
It's kinda time consuming to test this so I might not do only of them, but:
- float works
- fullscreen/maximize work but they affect a different window (??)
- monitor works

**Comment by MightyPlaza** [comment]
works for me on nested + git

**Comment by diniamo** [comment]
@MightyPlaza did you try with a special workspace?

**Comment by MightyPlaza** [comment]
`exec-once=[workspace special:terminal silent] foot`

**Comment by diniamo** [comment]
Huh it works with foot, just not alacritty

**Comment by diniamo** [comment]
Could it have something to do with my alacritty having a small startup delay?

**Comment by vaxerski** [comment]
doubt, maybe it forks?

**Comment by diniamo** [comment]
I have no idea how to confirm that, but it shouldn't

**Comment by leon-erd** [comment]
I also discovered some weird behaviour with alacritty and rules yesterday.
If I start it in the following way:
```
hyprctl dispatch "exec [float; size 33% 10%; move 30% 10%; tag noblur; noblur] alacritty -o window.opacity=0" && sleep 1 && hyprctl clients`
```
Then it has the correct tag and no blur behind it. However, if I switch back to the alacritty window where I executed the command above and run `hyprctl clients` again, the tag is gone. And when I switch back to the created alacritty window it also has blur again.

I can also confirm that the pid stays the same.
I don't know if this only affects alacritty but apparently its not limited to exec-once.
@diniamo can you try to reproduce my bug? If you can maybe change the title of this :)

**Comment by diniamo** [comment]
Yes I can reproduce your issue, however I doubt it's the same as mine. Can you reproduce mine?

**Comment by leon-erd** [comment]
Okay I will open another issue then.
Hm I just tried to reproduce your bug by putting `exec-once=[workspace special:tterminal silent] alacritty` in my hyprland.conf and it seems to work correctly.
`hyprctl dispatch togglespecialworkspace tterminal` brings me to the opened alacritty window

**Comment by vaxerski** [comment]
Hello there!

This issue has been closed, as we are moving from issues to discussions. More about this can be found [here](https://wiki.hyprland.org/Contributing-and-Debugging/Issue-Guidelines/) and in #9854.

Firstly, please make sure this issue is still relevant in latest Hyprland. If so, we ask you to open a [discussion](https://github.com/hyprwm/Hyprland/discussions) (please read the discussion guidelines first, in the pinned post)

Thank you for your understanding!
